receiver/prometheusreceiver/README.md

```diff
@@ -111,6 +111,7 @@ The prometheus receiver also supports additional top-level options:
 - **trim_metric_suffixes**: [**Experimental**] When set to true, this enables trimming unit and some counter type suffixes from metric names. For example, it would cause `singing_duration_seconds_total` to be trimmed to `singing_duration`. This can be useful when trying to restore the original metric names used in OpenTelemetry instrumentation. Defaults to false.
 - **use_start_time_metric**: When set to true, this enables retrieving the start time of all counter metrics from the process_start_time_seconds metric. This is only correct if all counters on that endpoint started after the process start time, and the process is the only actor exporting the metric after the process started. It should not be used in "exporters" which export counters that may have started before the process itself. Use only if you know what you are doing, as this may result in incorrect rate calculations. Defaults to false.
 - **start_time_metric_regex**: The regular expression for the start time metric, and is only applied when use_start_time_metric is enabled.  Defaults to process_start_time_seconds.
+- **enable_protobuf_negotiation**: When set to true, Prometheus receiver will try to negotiate the protobuf format first for scraping the target. This is considered experimental.
```

Comment:
Prometheus will enable two things by default eventually (I'd guess next year):
- try Protobuf negotiation first (since it's more efficient than text)
- allow native histograms (currently feature flag)

Actually the text format for native histograms is being worked on currently, so again eventually it will not matter what you set protobuf negotiation to.

Given the current code, to follow Prometheus, the default for `enable_protobuf_negotiation` option should be switched to `true` in otel collector as well when Prometheus makes the change.

The real "problem" will be native histograms being on by default. Since for a metric that has both classic and native buckets, we can only scrape one due to the name conflict (both result in the same metric name). So if we start enabling native histograms by default, suddenly the classic buckets would be gone, replaced by exponential buckets.

Reply:
So, we would want our eventual configuration to only include a knob for switching between fixed-bucket and exponential histogram, right?  Do we need to keep config for protobuf negotiation around long-term?  It sounds like we essentially just want to keep the default histogram format as fixed-bucket, rather than exponential

Reply:
> So, we would want our eventual configuration to only include a knob for switching between fixed-bucket and exponential histogram, right? Do we need to keep config for protobuf negotiation around long-term? It sounds like we essentially just want to keep the default histogram format as fixed-bucket, rather than exponential

I guess so. The question that comes up for Prometheus is different, because you can have the classic histograms beside the native histograms (as names are different). Still the issue is similar in that if we enable native histograms by default, you suddenly have a breaking change by switching from classic buckets to native buckets. And maybe we don't need an extra setting since you can set `scrape_classic_histograms` in the scrape config. In Prometheus that gets you the classic buckets **as well** and in otel collector it would get you the classic buckets **only**. At least in the my current implementation.

The whole thing only affects cases where the metric has both classic and native defined.

cc @beorn7

Reply:
Enabling protobuf negotiation can now be done using the prometheus scrape config.  See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30934.

Reply:
Will wait for  PR #30934  and rebase.

Reply:
I think this can be removed now